### PR TITLE
Add slack notifications

### PR DESCRIPTION
**Pull request by ayush-goyal**
Changes
----
- Add slack notifications sent to requisition creator when requisition status is changed
- If changes are requested, it will let the creator know via a message

### CODE REVIEW

File: expenses/config.py [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+
+def send_slack_notification(requisition):
+    # Checks if slack token is provide as env var
+    if slack_client.token and requisition.created_by.slack_id:
+        if requisition.status == "Submitted":
+            message = f"Thank you for submitting requisition {requisition.headline}! You will receive alerts from me when the status is changed."
+        elif requisition.status == "Pending Changes":
+            message = f"Requisition {requisition.headline} ({requisition}) has requested changes. Notes by reviewer: {requisition.approval_set.latest('id').notes}"
+        else:
+            message = f"Requisition {requisition.headline} ({requisition}) status updated to *{requisition.status}*"
```
Comment (evan10s) [inline expenses/config.py:20]: not sure but I think linking the requisition reference string and putting the headline in parentheses, like:

> Requisition 2020-HGT7-01 (Streamers for the Klaus atrium)

might be a better format this.

But the main thing is that linking the requisition reference string would be really helpful

File: expenses/config.py [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
 storage_client = storage.Client()
 bucket = storage_client.bucket(os.environ["GOOGLE_STORAGE_BUCKET"])
+
+slack_client = WebClient(token=os.environ.get('SLACK_API_TOKEN', ''))
+
+
+def send_slack_notification(requisition):
+    # Checks if slack token is provide as env var
+    if slack_client.token and requisition.created_by.slack_id:
+        if requisition.status == "Submitted":
```
Comment (evan10s) [inline expenses/config.py:15]: I don't love the direct string comparison in the event the status name ever changes, but it's not a huge deal

File: expenses/config.py (hunk 1 above)
Comment (ayush-goyal) [reply expenses/config.py:20]: Yeah the {requisition} prints the reference string, I can change the format to that though, bc its reversed

File: expenses/config.py (hunk 2 above)
Comment (ayush-goyal) [reply expenses/config.py]: I changed it to use the status enum, thanks for pointing this out

File: expenses/controllers/RequisitionController.py [hunk 3]
```diff
@@ -75,13 +79,16 @@ def update_requisition(cls, info, id, data):
...
+            status_changed = data.status != query.first().status
+
             # If project changed, recalculate project requisition id
             if int(data.project) != int(query.first().project.id):
                 id_max = Requisition.objects.filter(project=project).aggregate(Max('project_requisition_id'))
 
                 new_data = {
                     "project": project,
-                    "project_requisition_id": (id_max["project_requisition_id__max"] or 0) + 1,
+                    "project_requisitionr_id": (id_max["project_requisition_id__max"] or 0) + 1,
```
Comment (evan10s) [inline expenses/controllers/RequisitionController.py:91]: `project_requisition**r**_id`

typo?

Comment (ayush-goyal) [reply expenses/controllers/RequisitionController.py]: Ooh good catch just fixed

**Pull request merged**